internal/utils/collections.go

```diff
@@ -0,0 +1,21 @@
+package utils
+
+// Filter copies values from src to dst that match the
+// predicate fn and returns the count of copied values.
+// If the length of dst is less than the number of matching
+// values from src, iteration will break once dst has been
+// filled to its length.
+func Filter[T any](dst, src []T, fn func(T) bool) int {
```

Comment:
https://github.com/bobg/go-generics/blob/v1.7.2/slices/slices.go#L256 ?

Reply:
that... is freaky

Reply:
Personally, I prefer my version as it allows me to say "give me the first x matches only". However, I'll go with whatever is generally used by the team/project.

Reply:
(additionally, mine avoids a heap allocation)

Reply:
**(suggestion):** @senojj , @elbuo8 Can we consider using `slices.DeleteFunc` from stdlib (I think available from 1.22 or if not 1.23 that was released recently)? This way we'd have less code to maintain here. 

Implementation details:
```go
// from /go/libexec/src/slices/slices.go

// DeleteFunc removes any elements from s for which del returns true,
// returning the modified slice.
// DeleteFunc zeroes the elements between the new length and the original length.
func DeleteFunc[S ~[]E, E any](s S, del func(E) bool) S {
	i := IndexFunc(s, del)
	if i == -1 {
		return s
	}
	// Don't start copying elements until we find one to delete.
	for j := i + 1; j < len(s); j++ {
		if v := s[j]; !del(v) {
			s[i] = v
			i++
		}
	}
	clear(s[i:]) // zero/nil out the obsolete elements, for GC
	return s[:i]
}
```

Reply:
TIL about DeleteFunc.

Good call, thank you. This PR seems to bring a lot less value now than when I stated it. Perhaps just closing it would make sense at this time.